public/app/core/components/PageActionBar/PageActionBar.tsx

```diff
@@ -16,41 +17,59 @@ export interface Props {
     value?: string;
     getSortOptions?: () => Promise<SelectableValue[]>;
   };
+  favoritesCheckbox?: FavoritesCheckbox;
 }
 
-export default class PageActionBar extends PureComponent<Props> {
-  render() {
-    const {
-      searchQuery,
-      linkButton,
-      setSearchQuery,
-      target,
-      placeholder = 'Search by name or type',
-      sortPicker,
-    } = this.props;
-    const linkProps: Omit<Parameters<typeof LinkButton>[0], 'children'> = {
-      href: linkButton?.href,
-      disabled: linkButton?.disabled,
-    };
-
-    if (target) {
-      linkProps.target = target;
-    }
+export default function PageActionBar({
```

Comment:
I have rewritten this component as a function component to be able to use `getStyles` and avoid the hardcoded className `page-action-bar`. The only changes are that style change and the new `favoritesCheckbox` parameter.